x-pack/legacy/plugins/siem/public/components/alerts_viewer/histogram_configs.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as i18n from './translations';
+import { MatrixHistogramOption } from '../matrix_histogram/types';
+import { HistogramType } from '../../graphql/types';
+
+export const alertsStackByOptions: MatrixHistogramOption[] = [
+  {
+    text: 'event.category',
+    value: 'event.category',
+  },
+  {
+    text: 'event.module',
+    value: 'event.module',
+  },
+];
+
+export const histogramConfigs = {
+  defaultStackByOption: alertsStackByOptions[1],
```

Comment:
Since the `defaultStackByOption` is initialized via array index:

```
defaultStackByOption: alertsStackByOptions[1],
``` 

another developer may, for example, add the following entry:

```
  {
    text: 'event.action',
    value: 'event.action',
  },
```

to `alertsStackByOptions`, which when added alphabetically, would look like this:

```
export const alertsStackByOptions: MatrixHistogramOption[] = [
  {
    text: 'event.action',
    value: 'event.action',
  },
  {
    text: 'event.category',
    value: 'event.category',
  },
  {
    text: 'event.module',
    value: 'event.module',
  },
];
```

In this example, the developer made a reasonable decision to add the new entry in alphabetical order, but as a result, the `defaultStackByOption` will reference `event.category` instead of `event.module`, unless the developer noticed the `alertsStackByOptions[1]` below it.

Consider exploring alternatives for providing (type safe) initialization of the `defaultStackByOption`, and applying that alternative to all the histograms that have a `defaultStackByOption`.

Reply:
Good point, now I see why you have the alternatives on overview page.